framework/src/utils/MooseEnum.C

```diff
@@ -105,6 +105,10 @@ MooseEnum::operator==(const char * name) const
   std::string upper(name);
   std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
 
+  if (std::find(_names.begin(), _names.end(), upper) == _names.end() && (upper != ""))
+    if (_out_of_range_index == 0)     // Are out of range values allowed?
```

Comment:
This whole check is incredibly expensive.  While you shouldn't be using string comparisons in your computational functions (because they are already slow)... maybe this new code should only be in debug mode or in an assert?